### PR TITLE
Add crimson metric

### PR DESCRIPTION
**Pull request by siavashraissi**
I added the CRIMSON metric to RadEval. **CRIMSON** is a recently released LLM-based metric from the Rajpurkar Lab that incorporates clinical context (patient age, indication, clinical guideline), breaks down individual findings into fine-grained error categories, and scales the clinical significance of errors to produce a final weighted score (Details located at https://arxiv.org/abs/2603.06183, https://github.com/rajpurkarlab/CRIMSON). 

Similar to GREEN and Mammo-GREEN, CRIMSON was implemented into RadEval with the option to either run the metric using OpenAI's API, or a finetuned MedGemma model specifically trained to generate CRIMSON scores (https://huggingface.co/CRIMSONScore/medgemma-4b-it-crimson). The package's functionality was validated using new tests (`tests/test_crimson.py`). `README.md,` `docs/configuration.md`,  and `docs/metrics.md` were updated, respectively, to describe this new functionality. 

If you have any questions about CRIMSON or its implementation, please feel free to reach out to siavash_raissi@hms.harvard.edu.

### CODE REVIEW

**Comment by jbdel** [comment]
Hi @siavashraissi, thanks for this contribution! I'm also super happy you reference radeval in the crimson paper.
Would be super cool to to partner with rexrank? perhaps rexrank could rely on radeval for full transparency on how the metrics are computed? feel free to shoot me an email if need be.

Now, about the PR:

I have integrated your metric into the current codebase (the repo has been significantly restructured since your fork, so a direct merge wasn't feasible). Your code is now live.

**What was kept as-is:**
- `prompt_parts.py` -- the composable prompt design is excellent
- `_calculate_crimson()` scoring logic -- untouched
- `CRIMSONScore` / `CRIMSON` class API
- Your test structure and mock data

**What I adjusted during integration:**
- **Lazy imports**: `torch`, `openai`, `transformers` are now only imported inside the relevant backend branch, so users who don't enable CRIMSON don't need these installed
- **JSON parsing**: Added robustness for markdown fences and trailing commas (common LLM output quirks)
- **Retry logic**: Added tenacity retry (3 attempts, exponential backoff) on API calls
- **Response validation**: Added a check on the LLM output structure before passing to `_calculate_crimson()`
- **Logging**: Replaced `print()` with `logging.getLogger(__name__)`
- **Progress tracking**: Added `on_sample_done` callback for rich progress bars

**All of your original tests pass with these changes (80 tests passing across the full suite, including all CRIMSON unit and RadEval integration tests with mocked backends).**

The default backend is HuggingFace MedGemma (`crimson_api="hf"`), matching `CRIMSONScore`'s own default.

usage:
```python
from RadEval import RadEval

refs = ["Moderate right pleural effusion. No pneumothorax."]
hyps = ["Small right pleural effusion. No pneumothorax."]

# HuggingFace MedGemma (default, runs locally on GPU)
evaluator = RadEval(do_crimson=True)

# Or use OpenAI API
evaluator = RadEval(
    do_crimson=True,
    crimson_api="openai",
    crimson_api_key="sk-...",
)

results = evaluator(refs=refs, hyps=hyps)
print(results["crimson"])  # CRIMSON score (range [-1, 1])
```
I'm closing this PR since the code has been integrated directly, but full credit to you in the commit message. Let me know if anything looks off!
